pelican/readers.py

```diff
@@ -246,8 +246,13 @@ class MarkdownReader(BaseReader):
 
     def __init__(self, *args, **kwargs):
         super(MarkdownReader, self).__init__(*args, **kwargs)
-        self.extensions = self.settings['MD_EXTENSIONS']
-        self.extensions.setdefault('markdown.extensions.meta', {})
+        # make sure 'extension_configs' exists and
+        # and either way 'markdown.extensions.meta' must be in there
+        settings = self.settings['MARKDOWN_SETTINGS']
+        settings.setdefault('extension_configs', {})
+        settings['extension_configs'].setdefault(
+            'markdown.extensions.meta', {})
+        settings['extensions'] = list(settings['extension_configs'].keys())
```

Comment:
You're right, fixed it.
Now it is also possible to load custom extensions e.g.:
```
from markdown.extensions import Extension

class MyExtension(Extension):
    def extendMarkdown(self, md, md_globals):
        print("run MyExtension")

MARKDOWN_SETTINGS = {
    'extensions': ['markdown.extensions.headerid', MyExtension()],
    'extension_configs': {
        'markdown.extensions.extra': {},
        'markdown.extensions.headerid': {'level': 1},
        'markdown.extensions.codehilite': {'css_class': 'highlight', 'linenums': True},
    }
}
```